detox/src/client/AsyncWebSocket.js

```diff
@@ -124,6 +129,14 @@ class AsyncWebSocket {
     return _.some(this.inFlightPromises, p => p.message.type !== 'currentStatus');
   }
 
+  hasMultiplePendingActions() {
+    const remainingAfterRemovingCheckIdleMessages = _.filter(this.inFlightPromises, p =>
```

Comment:
Not correct. You should also exclude `captureViewHierarchy`, `setRecordingState`, `takeScreenshot`.

Also, you should never block `cleanup` action, I guess. @alon-ha , could you advise too?

Reply:
Naming... please use a shorter variable name here 😄